lib/replicator.js

```diff
@@ -386,6 +386,20 @@ class Peer {
     if (drop) this._unclearLocalRange(start, length)
     else this._clearLocalRange(start, length)
 
+    // TODO: consider also adding early-returns on the drop===true case
+    if (!drop) {
+      if (this._remoteContiguousLength >= start + length) return
+
+      // TODO: consider doing more efficiently. Maybe a function directly in the bitfield logic?
+      let nrFound = 0
+      for (let i = start; i < start + length; i++) {
+        if (!this.remoteBitfield.get(i)) break
+        nrFound++
+      }
```

Comment:
just search for the next false (there is an fast api for that). if the next false is < start+length, send, otherwise no